Factory/DiactorosFactory.php

```diff
@@ -0,0 +1,159 @@
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createRequest(Request $symfonyRequest)
+    {
+        $server = DiactorosRequestFactory::normalizeServer($symfonyRequest->server->all());
+        $headers = $symfonyRequest->headers->all();
+
+        try {
+            $body = new DiactorosStream($symfonyRequest->getContent(true));
+        } catch (\LogicException $e) {
+            $body = new DiactorosStream('php://temp', 'wb+');
+            $body->write($symfonyRequest->getContent());
+        }
+
+        $request = new ServerRequest(
+            $server,
+            DiactorosRequestFactory::normalizeFiles($this->getFiles($symfonyRequest->files->all())),
+            DiactorosRequestFactory::marshalUriFromServer($server, $headers),
+            $symfonyRequest->getMethod(),
+            $body,
+            $headers
+        );
+
+        $request = $request
+            ->withCookieParams($symfonyRequest->cookies->all())
+            ->withQueryParams($symfonyRequest->query->all())
+            ->withParsedBody($symfonyRequest->request->all())
+        ;
+
+        foreach ($symfonyRequest->attributes->all() as $key => $value) {
+            $request = $request->withAttribute($key, $value);
+        }
+
+        return $request;
+    }
+
+    /**
+     * Converts Symfony uploaded files array to the PSR one.
+     *
+     * @param array $uploadedFiles
+     *
+     * @return array
+     */
+    private function getFiles(array $uploadedFiles)
+    {
+        $files = array();
+
+        foreach ($uploadedFiles as $key => $value) {
+            if ($value instanceof UploadedFile) {
+                $files[$key] = $this->createUploadedFile($value);
+            } else {
+                $files[$key] = $this->getFiles($value);
+            }
+        }
+
+        return $files;
+    }
+
+    /**
+     * Creates a PSR-7 UploadedFile instance from a Symfony one.
+     *
+     * @param UploadedFile $symfonyUploadedFile
+     *
+     * @return UploadedFileInterface
+     */
+    private function createUploadedFile(UploadedFile $symfonyUploadedFile)
+    {
+        return new DiactorosUploadedFile(
+            $symfonyUploadedFile->getRealPath(),
+            $symfonyUploadedFile->getSize(),
+            $symfonyUploadedFile->getError(),
+            $symfonyUploadedFile->getClientOriginalName(),
+            $symfonyUploadedFile->getClientMimeType()
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createResponse(Response $symfonyResponse)
+    {
+        $stream = new DiactorosStream('php://temp', 'wb+');
+        if ($symfonyResponse instanceof StreamedResponse) {
+            ob_start(function ($buffer) use ($stream) {
+                $stream->write($buffer);
+
+                return false;
+            });
+
+            $symfonyResponse->sendContent();
+            ob_end_clean();
```

Comment:
Because I use `php://temp`, after 2Mo the data will be stored in a temporary file on disk.